src/main/scala/com/databricks/labs/overwatch/pipeline/PipelineFunctions.scala

```diff
@@ -90,9 +90,11 @@ object PipelineFunctions {
     val maxCoreCount = pipeline.getConfig.intelligentScaling.maximumCores
     val baseCoreCount = pipeline.getConfig.intelligentScaling.minimumCores
     val userCoeffMultiplier = pipeline.getConfig.intelligentScaling.coeff
+    val coresPerWorker = pipeline.getCoresPerWorker
     if (pipeline.getConfig.intelligentScaling.enabled) {
-      val nodeCountUpperBound = Math.floor(maxCoreCount / pipeline.getCoresPerWorker).toInt
-      val newNodeCount = Math.min(Math.max(baseCoreCount, Math.ceil(baseCoreCount * scaleCoefficient * userCoeffMultiplier).toInt), nodeCountUpperBound)
+      val nodeCountUpperBound = Math.floor(maxCoreCount / coresPerWorker).toInt
+      val newCoreCount = Math.min(Math.max(baseCoreCount, Math.ceil(baseCoreCount * scaleCoefficient * userCoeffMultiplier).toInt), maxCoreCount)
+      val newNodeCount = Math.min(Math.floor(newCoreCount / coresPerWorker), nodeCountUpperBound).toInt
```

Comment:
This logic was incorrect. I was accidentally comparing nodes to cores and thus it was always selecting the max node count. This corrects that.